### DISCUSSION TITLE
Steps from 3rd at SDR to FIRST™ at 2024LAKE

### DISCUSSION THREAD

**Discussion by GalexY727** [Patribots4738/Crescendo2024#281] [General]
As always, feel free to edit this message and add more if you have other ideas
## **This list is in order.**
- [x] Figure out if we can/should do 2 color sensors 
  - (we can't have two on the same i2c bus without a multiplexer, they have the same address)
- [x] figure out if we want to tune the focus of the LL3G at home or at comp and discuss what method we need to use to seal the lens.
- [x] Make sure obj detection on LL3 is good, perhaps better than that of the 2+
- [x] test left bumper double-amping :)
  - [ ] falling edge intake detection for auto-stopping
- [x] put up april tags :( `red alliance best alliance`
- [x] Test 4481 inspired wing autos (with no notes???)
  - If this new auto works, that's really nice
  - but there should not be more than 20-30 mins tuning it if it doesn't work.
- [x] use our bad note(s) to verify interpolation map isn't tweaking (no more than 15mins)
- [x] make sure (if we had a good note) we wouldn't cut it when we try to amp (b)
  - this might prompt moving the color sensor and utilizing the proximity detector on it, which might be really useful but idk where to put all that
- [ ] make it so that the note moves back just a touch to never have to worry about the note shooting out the shooter if we have an excited intake
- [x] enable auto shooter speeds if intaking via driver holding left trigger but close to speaker (within spike mark area)
  - [x] make work with operator alternative button
- [ ] find a way to make obj detection better and smoother in autos
- [ ] force pivot down on b press even if the right stick was toggled
- [ ] maybe tune interpolation map on field using a NT angle++ variable *(sketchy)
- [ ] perhaps re-tune hdc for heading keeping in mind a +- 3 degree allowable offset

**Comment by jacueblol** [comment]
I recommend tuning the focus here and sealing it, however, if the sealing is not super strong, do it at comp.
